ouroboros-network/protocol-tests/Ouroboros/Network/Protocol/KeepAlive/Examples.hs

```diff
@@ -0,0 +1,43 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Ouroboros.Network.Protocol.KeepAlive.Examples where
+
+import           Data.List (cycle)
+
+import           Ouroboros.Network.Protocol.KeepAlive.Server
+import           Ouroboros.Network.Protocol.KeepAlive.Client
+
+
+-- | A client which applies a function whenever it receives
+-- 'MsgKeepAliveResponse' and returns the result.
+--
+keepAliveClientApply :: forall acc m. Monad m
+                    => (acc -> acc)
+                    -> acc
+                    -> Int
+                    -- ^ number of rounds 
```

Comment:
'number of rounds" may be missinterpreted, given the crypto context of this project. 
 Its this really "count of number of requests"?